### PR TITLE
Fix windows path check on rust linter

### PR DESCRIPTION
**Pull request by daa84**
Fix for #724 
I'm not sure it is good solution, but for me it works.

### CODE REVIEW

File: autoload/ale/handlers/rust.vim [hunk 1]
```diff
@@ -46,9 +46,11 @@ function! ale#handlers#rust#HandleRustErrorsForFile(buffer, full_filename, lines
         endif
 
         for l:span in l:error.spans
+            " substitue used for window file paths
+            let l:file_name = substitute(l:span.file_name, '\\', '\\\\', 'g')
             if (
             \   l:span.is_primary
-            \   && (a:full_filename =~ (l:span.file_name . '$') || l:span.file_name ==# '<anon>')
+            \   && (a:full_filename =~ (l:file_name . '$') || l:span.file_name ==# '<anon>')
```
Comment (w0rp) [inline autoload/ale/handlers/rust.vim]: Could you try using `ale#path#IsBufferPath` for this instead? I'm trying to use the one function for handling everything. That's step 1 towards supporting setting errors from other files in quickfix. Step two is outputting `'filename': whatever` in the Dictionary objects, and then checking the filename returned automatically.

**Comment by daa84** [comment]
I replace code with this function and for me it works but find problem with this when run tests on windows - tests does not pass because ale#test#SetFilename('src/playpen.rs') converts "/" to "\" in this case

**Comment by w0rp** [comment]
@daa84 That's okay. The tests don't run on Windows at the moment. I'm planning on running the tests on AppVeyor eventually.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**